packages/components/src/GeometryPanel.component.svelte

```diff
@@ -0,0 +1,209 @@
+<svelte:options customElement="geometry-panel" />
+
+<script lang="typescript">
+  import { onMount, onDestroy } from 'svelte';
+  import Section from './Section.component.svelte';
+  import TypedTextBox from './TypedTextBox.component.svelte';
+  import { Events } from './constants/events';
+  import { ButtonProps, GUIEvent } from './types';
+
+  export let attributes = '[]';
+
+  let selectedGeometry = undefined;
+
+  function loadPickedObject(event: CustomEvent<GUIEvent>) {
+    const target = event.detail.target.split('_');
+
+    if (target[0] !== 'geometry') return;
+
+    selectedGeometry = event.detail.props;
+
+    event.stopPropagation();
+  }
+
+  function clearSelectedGeometry() {
+    selectedGeometry = undefined;
+
+    window.dispatchEvent(
+      new CustomEvent<GUIEvent<ButtonProps>>(Events.BUTTON_CLICK, {
+        bubbles: true,
+        detail: {
+          target: 'geometry',
+          props: {
+            command: 'clear'
+          }
+        }
+      })
+    );
+  }
+
+  onMount(() => {
+    window.addEventListener(Events.PICK_OBJECT, loadPickedObject, {
+      capture: true
+    });
+  });
+
+  onDestroy(() => {
+    window.removeEventListener(Events.PICK_OBJECT, loadPickedObject, {
+      capture: true
+    });
+  });
+
+  let attributeMap = new Map(
+    JSON.parse(attributes ?? '[]').map(item => [item.name, item])
+  );
+</script>
+
+<Section>
+  <div slot="header" class="Viewer-GeometrySelector__title">
+    <svg
+      width="24"
+      height="24"
+      viewBox="0 0 304 296"
+      fill="none"
+      xmlns="http://www.w3.org/2000/svg"
+    >
+      <rect
+        x="45"
+        y="109"
+        width="96"
+        height="96"
+        stroke="var(--viewer-text-disabled)"
+        stroke-width="14"
+      />
+      <path
+        d="M90.1724 137.5L143 46L195.828 137.5H90.1724Z"
+        stroke="var(--viewer-text-disabled)"
+        stroke-width="14"
+      />
+      <circle
+        cx="212"
+        cy="203"
+        r="48"
+        stroke="var(--viewer-text-disabled)"
+        stroke-width="14"
+      />
+    </svg>
+    Geometry
+  </div>
+  <div class="Viewer-GeometrySelector__content" slot="content">
+    {#if selectedGeometry !== undefined}
+      <div class="Viewer-GeometrySelector__pickInfo">
+        {#each Object.entries(selectedGeometry).sort( (a, b) => (a[0] > b[0] ? 1 : b[0] > a[0] ? -1 : 0) ) as [key, value]}

```

Comment:
Can we extract this comparator function to its own module with a couple of tests? Even switching to a named function will make things more readable